### DISCUSSION TITLE
Attribute modifiers for array-casted fields

### DISCUSSION THREAD

**Discussion by zlodes** [owen-it/laravel-auditing#796] [Q&A] [answered]
| Q                 | A
| ----------------- | ---
| Bug?              | no
| New Feature?      | yes
| Framework         | Laravel
| Framework version | 7.29.0
| Package version   | 11.0.0
| PHP version       | 7.4.10

It will be very handy to have attribute modifiers for fields casted to array.

Example:

```php
protected $casts = [
    'credentials' => 'array', // jsonb
];

protected $attributeModifiers = [
    'credentials.password' => RightRedactor::class,
];
```

### Before modifications:
```php
[
    'email' => 'alice@example.com',
    'password' => 'Pa$$w0rd',
]
```

### After modifications:
```php
[
    'email' => 'alice@example.com',
    'password' => 'Pa******',
]
```

**Comment by parallels999** [comment, marked as answer]
Seems not possible because it only has attributes names, you could use `transformAudit($data)` for changing that
https://github.com/owen-it/laravel-auditing/blob/5f86086f0465c5701e065771096d0cd9e338a6c9/src/Auditable.php#L346-L349
Also you could make a PR for adding that feature
